### PR TITLE
feat: add support for shutdownGracePeriod and shutdownGracePeriodCriticalPods

### PR DESCRIPTION
**Pull request by kahirokunn**
`ALB (target-type: instance) -> nodeport -> pod` or `NLB -> pod` configuration is assumed.

For example, if you have an EKS cluster with vxlan etc. instead of IPAM mode: eni, this is the best practice.

If you have multiple Nodes that only have DaemonSet, you will naturally want to terminate them.
However, if you terminate them, the TCP connection from ELB -> Node to be terminated will be broken.

Therefore, the Node is removed from the target group in advance, but in the case of ALB, the currently connected connection can be maintained for up to 3600 seconds.
In other words, the survival period of a Node must be longer than the time to maintain the current connection.

Also, for example, if you are using NLB, you may want to keep a Node alive for a longer period of time if it has been used beyond the 3600 second barrier.

### CODE REVIEW

**Comment by bwagner5** [comment]
Thanks for opening this PR! I haven't looked at this in a while, but last time I did, Bottlerocket didn't support this kubelet configuration and Amazon Linux 2 had an issue with the version of systemd in-use to support kubelet graceful shutdown properly.  

Are you able to implement the aws/karpenter logic and see if kubelet graceful node shutdown works on AL2 and Bottlerocket now?

**Comment by kahirokunn** [comment]
@bwagner5 
Although still Open PR, there are traces of support for ShutdownGracePeriod and ShutdownGracePeriodCriticalPods via userData in bottlerocket.
So, if this one is merged, I think Karpenter could be supported as well.
https://github.com/bottlerocket-os/bottlerocket/pull/2930

**Comment by bwagner5** [comment]
Are you able to open a PR in github.com/aws/karpenter to implement the behavior for BR so we can have it supported when that PR is merged in BR?

**Comment by njtran** [comment]
@kahirokunn any update?

**Comment by stmcginnis** [comment]
Related to https://github.com/bottlerocket-os/bottlerocket/pull/2930 - Bottlerocket 1.14.0 should have these settings in place to set graceful shutdown behavior. 1.14.0 should be available very shortly.

**Comment by kahirokunn** [comment]
https://github.com/aws/karpenter-core/pull/248#issuecomment-1540643087

Yes, I am waiting for this.

**Comment by bwagner5** [comment]
Instead of adding this to the kubelet configuration in the CRD, can you open a PR in aws/karpenter to just add it to the Bottlerocket configuration block https://github.com/aws/karpenter/blob/main/pkg/providers/amifamily/bootstrap/bottlerocketsettings.go#L45

Since these parameters aren't useful for scheduling purposes, I think it makes sense to just to allow for Bottlerocket for now specified in the user-data toml (especially since AL2 doesn't support this parameter because of the old version of systemd shipped with AL2).
